app/stac_api/validators.py

```diff
@@ -168,6 +171,46 @@ def validate_link_rel(value):
         )
 
 
+def validate_text_to_geometry(text_geometry):
+    '''
+    https://jira.swisstopo.ch/browse/BGDIINF_SB-1650
+    A validator function that tests, if a text can be transformed to a geometry.
+    The text is either a bbox or WKT.
+
+    an extent in either WGS84 or LV95, in the form "(xmin, ymin, xmax, ymax)" where x is easting
+    a WKT defintion of a polygon in the form POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))
+
+    Args:
+        text_geometry: The text to be transformed into a geometry
+    Returns:
+        the text_geometry
+    Raises:
+        ValidationError: About that the text_geometry is not valid
+    '''
+    if not text_geometry:
+        message = "The geometry is a required field: %s"
+        params = {'error': 'empty field'}
+        logger.error(message, params)
+        raise ValidationError(_(message), params=params, code='invalid') from None
+    # is the input WKT
+    try:
+        geos_geometry = GEOSGeometry(text_geometry)
+        validate_geometry(geos_geometry)
+        return geos_geometry
+    except (ValueError, ValidationError, IndexError, GDALException, GEOSException) as error:
+        message = "The text as WKT could not be transformed into a geometry: %s " % error
+    # is the input a bbox
+    try:
+        text_geometry = text_geometry.replace('(', '')
+        text_geometry = text_geometry.replace(')', '')
+        return validate_geometry(geometry_from_bbox(text_geometry))
+    except (ValueError, ValidationError, IndexError, GDALException) as error:
+        message += "The text as bbox could not be transformed into a geometry: %s"
+        params = {'error': error}
+        logger.error(message, params)
+        raise ValidationError(_(message), params=params, code='invalid') from None
```

Comment:
Error handling could be improved, and the messages formatting is not correct. I proposed to do so:
```python
    errors = []
    try:
        geos_geometry = GEOSGeometry(text_geometry)
        validate_geometry(geos_geometry)
        return geos_geometry
    except (ValueError, ValidationError, IndexError, GDALException, GEOSException) as error:
        message = "The text as WKT could not be transformed into a geometry: %(error)s"
        params = {'error': error}
        logger.warning(message, params)
        errors.append(ValidationError(_(message), params=params, code='invalid'))
    # is the input a bbox
    try:
        text_geometry = text_geometry.replace('(', '')
        text_geometry = text_geometry.replace(')', '')
        return validate_geometry(geometry_from_bbox(text_geometry))
    except (ValueError, ValidationError, IndexError, GDALException) as error:
        message = "The text as bbox could not be transformed into a geometry: %(error)s"
        params = {'error': error}
        logger.error(message, params)
        errors.append(ValidationError(_(message), params=params, code='invalid'))
        raise ValidationError(errors) from None
```
See https://docs.djangoproject.com/en/3.2/ref/forms/validation/#raising-multiple-errors

Below in line 234 I also see a mistake in the error message, it should be: `message = "The geometry is not valid: %(error)s"`